engines/gob/inter_v7.cpp

```diff
@@ -263,6 +352,218 @@ void Inter_v7::o7_loadFunctions() {
+	if (isPattern) {
+		files = _vm->_saveLoad->getFilesMatchingPattern(file.c_str());
+		debugC(2, kDebugFileIO, "Delete file matching pattern \"%s\" (%d matching file(s))", file.c_str(), files.size());
+		for (const Common::String &matchingFile : files)
+			debugC(5, kDebugFileIO, "Matching file: %s", matchingFile.c_str());
+	}
+	else {
+		files.push_back(file);
+		debugC(2, kDebugFileIO, "Delete file \"%s\"", file.c_str());
+	}
+
+	if (_vm->getGameType() == kGameTypeAdibou2
+		&& isPattern
+		&& file.hasPrefix("DATA\\??????")) {
+		// WORKAROUND a bug in original game: files APPLI_<N>.INF and CRITE_<N>.INF should not be deleted when removing character <N>
+		// Those files contain *application <N>* data, not "character <N>" data
+		for (Common::List<Common::String>::iterator it = files.begin(); it != files.end(); ++it) {
+			if (it->matchString("DATA\\\\appli_??.inf", true) || it->matchString("DATA\\\\crite_??.inf", true)) {
+				debugC(2, kDebugFileIO, "o7_deleteFile: ignoring deletion of file \"%s\" when processing pattern %s (delete character bug workaround)",
+					   it->c_str(),
+					   file.c_str());
+				it = files.reverse_erase(it);
+			}
+		}
+	}
+
+	for (const Common::String &fileToDelete : files) {
+		SaveLoad::SaveMode mode = _vm->_saveLoad->getSaveMode(fileToDelete.c_str());
+		if (mode == SaveLoad::kSaveModeSave) {
+			if (!_vm->_saveLoad->deleteFile(fileToDelete.c_str())) {
+				warning("Cannot delete file \"%s\"", fileToDelete.c_str());
+			}
+		} else if (mode == SaveLoad::kSaveModeNone)
+			warning("Attempted to delete file \"%s\"", fileToDelete.c_str());
+	}
+}
+
+void Inter_v7::o7_moveFile() {
+	Common::String path1 = _vm->_game->_script->evalString();
+	Common::String path2 = _vm->_game->_script->evalString();
+
+	Common::String file1 = getFile(path1.c_str(), false);
+	Common::String file2 = getFile(path2.c_str(), false);
+
+	if (file1.equalsIgnoreCase(file2)) {
+		warning("o7_moveFile(): \"%s\" == \"%s\"", path1.c_str(), path2.c_str());
+		return;
+	}
+
+	copyFile(file1, file2);
+	SaveLoad::SaveMode mode = _vm->_saveLoad->getSaveMode(file1.c_str());
+	if (mode == SaveLoad::kSaveModeSave) {
+		_vm->_saveLoad->deleteFile(file1.c_str());
+	} else if (mode == SaveLoad::kSaveModeNone)
+		warning("Attempted to delete file \"%s\" while moving it to \"%s\"", file1.c_str(), file2.c_str());
+}
+
+
+void Inter_v7::o7_initScreen() {
+	// TODO: continue implementation
+	int16 offY;
+	int16 videoMode;
+	int16 width, height;
+
+	offY = _vm->_game->_script->readInt16();
+
+	videoMode = offY & 0xFF;
+	offY = (offY >> 8) & 0xFF;
+
+	width = _vm->_game->_script->readValExpr();
+	height = _vm->_game->_script->readValExpr();
+
+	if (videoMode == 0)
+		videoMode = 0x14;
+
+	_vm->_video->clearScreen();
+
+	if (videoMode == 0x13) {
+
+		if (width == -1)
+			width = 320;
+		if (height == -1)
+			height = 200;
+
+		_vm->_width = 320;
+		_vm->_height = 200;
+
+		_vm->_video->setSize();
+
+	} else if (_vm->_global->_videoMode == 0x13) {
```

Comment:
0x13 or 0x14?

Reply:
I do not know, but I would say 0x13 is correct ; this part has been copied from a base class (inter_v5). (it is not the same "videomode" that is checked in the two "if" conditions)